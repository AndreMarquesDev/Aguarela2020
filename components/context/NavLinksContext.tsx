import { createContext, Context } from 'react';

export interface NavLinksContextProps {
    setNavHeight?: React.Dispatch<React.SetStateAction<number>>;
}

const NavLinksContext: Context<NavLinksContextProps> = createContext(null);

export default NavLinksContext;
