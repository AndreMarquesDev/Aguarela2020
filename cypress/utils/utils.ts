import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
