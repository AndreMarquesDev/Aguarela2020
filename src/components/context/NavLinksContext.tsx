import type { Context } from 'react';
import { createContext } from 'react';

export type NavLinksContextProps = {
    isMenuOpen: boolean;
    toggleMenu: React.Dispatch<React.SetStateAction<boolean>>;
    setNavHeight?: React.Dispatch<React.SetStateAction<number>>;
};

// @ts-ignore
export const NavLinksContext: Context<NavLinksContextProps> = createContext(null);
