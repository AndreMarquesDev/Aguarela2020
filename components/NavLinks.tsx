/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { FC } from 'react';
import Link from 'next/link';
import classNames from 'classnames';
import { getPrismicText } from '../utils/generic';
import breakpoints from '../styles/breakpoints';

export interface NavLink {
    spans: string[];
    text: string;
    type: string;
}

interface NavLinksProps {
    navLinks: Array<[string, NavLink[]]>;
    currentRoute: string;
    language: string;
}

const NavLinks: FC<NavLinksProps> = props => {
    const {
        navLinks,
        currentRoute,
        language,
    } = props;

    const linkStyles = (link: string): string => (
        classNames('link',
            currentRoute === `/${link}` && 'active')
    );

    return (
        <ul className="genericMargins">
            {navLinks.map(link => {
                const linkType = link[0];
                const linkText = getPrismicText(link[1]);

                return (
                    <li key={linkType}>
                        <Link href={`/${language}/${linkType}`} prefetch={false}>
                            <a className={linkStyles(linkType)}>
                                {linkText}
                            </a>
                        </Link>
                    </li>
                );
            })}

            <style jsx>
                {`
                    ul {
                        display: flex;
                        flex-wrap: wrap;
                        justify-content: center;
                        list-style: none;
                        margin-bottom: 0;
                    }

                    li {
                        margin: 0 10rem 15rem;

                        &:last-child {
                            margin-bottom: 30rem;

                        }

                        a {
                            font-size: 24rem;
                            text-transform: uppercase;
                        }

                    }

                    @media (max-width: ${breakpoints.mobile}) {
                        li {
                            margin: 0 5rem 10rem;

                            &:last-child {
                                margin-bottom: 15rem;

                            }
                        }
                    }
                `}
            </style>
        </ul>
    );
};

export default NavLinks;
