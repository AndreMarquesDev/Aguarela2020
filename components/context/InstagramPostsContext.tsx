import { createContext, Context } from 'react';
import { IInstagramPost } from '../../utils/generic';

export interface IInstagramPostsContext {
    posts: IInstagramPost[];
}

const InstagramPostsContext: Context<IInstagramPostsContext> = createContext(null);

export default InstagramPostsContext;
