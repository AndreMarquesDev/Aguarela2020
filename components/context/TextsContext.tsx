import { createContext, Context } from 'react';
import { TextsInterface, textsPt } from '../../utils/texts';

export interface TextsContextProps {
    texts: TextsInterface;
}

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
