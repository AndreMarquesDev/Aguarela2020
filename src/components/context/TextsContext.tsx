import type { Context } from 'react';
import type { TextsInterface } from '../../utils/texts';
import { createContext } from 'react';
import { textsPt } from '../../utils/texts';

export type TextsContextProps = {
    texts: TextsInterface;
};

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
