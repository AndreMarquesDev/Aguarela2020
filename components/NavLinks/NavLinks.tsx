import React, { useContext } from 'react';
import Link from 'next/link';
import classNames from 'classnames';
import { pagesMap } from '../../utils/pages';
import { TextsContext } from '../context/TextsContext';
import { LanguageButton } from '../LanguageButton/LanguageButton';
import { Locale } from '../../utils/locales';

export interface NavLinksProps {
    currentRoute: string;
    language: Locale;
    isMobile: boolean;
    isMenuOpen: boolean;
}

export const NavLinks = ({
    currentRoute,
    language,
    isMobile,
    isMenuOpen,
}: NavLinksProps): JSX.Element => {
    const navLinks = pagesMap.filter(page => page !== 'homepage');
    const { texts } = useContext(TextsContext);

    const linksContainerStyles = classNames(
        isMobile && 'mobileLayout',
        isMenuOpen && isMobile && 'menuOpen'
    );
    const linkStyles = (link: string): string =>
        classNames(currentRoute.includes(link) && 'active');

    return (
        <ul className={linksContainerStyles}>
            {navLinks.map(link => (
                <li key={language + link}>
                    <Link href={`/${language}/${link}`} prefetch={false}>
                        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                        <a className={linkStyles(link)}>{texts[`${link}`]}</a>
                    </Link>
                </li>
            ))}

            <li className={classNames('languageButton', isMobile && 'white')}>
                <LanguageButton />
            </li>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    ul {
                        display: flex;

                        &.mobileLayout {
                            display: none;

                            li {
                                a {
                                    color: $white;
                                    margin-bottom: 25rem;
                                }
                            }
                        }

                        &.menuOpen {
                            width: 100%;
                            height: 100vh;
                            position: fixed;
                            display: flex;
                            flex-direction: column;
                            align-items: center;
                            justify-content: center;
                            top: 0;
                            left: 0;
                            z-index: 1;
                            background: $purple;
                        }
                    }

                    li {
                        margin: 0 25rem;

                        &:hover a,
                        .active {
                            color: $purple;
                        }

                        @include phablet {
                            margin: 0 calc(12.5rem / 2);
                        }

                        a {
                            display: inline-block;
                            @include fontS($textTransform: uppercase);
                            transition: color 0.15s linear;
                        }

                        &.languageButton {
                            @include fontXXS($textTransform: uppercase);
                            align-self: center;
                            margin-top: 5rem;
                            margin-left: 0;

                            &:hover {
                                color: $purple;
                            }

                            &.white {
                                @include fontXS($white, uppercase);
                                margin-top: 0;
                            }

                            @include phablet {
                                margin: 5rem calc(12.5rem / 2) 0;
                            }
                        }
                    }
                `}
            </style>
        </ul>
    );
};
