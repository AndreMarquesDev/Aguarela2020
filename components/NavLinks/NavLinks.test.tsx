import '@testing-library/jest-dom';
import React from 'react';
import { render, screen, RenderResult } from '@testing-library/react';
import { NavLinks, NavLinksProps } from './NavLinks';
import { NavLinksContext } from '../context/NavLinksContext';
import { textsEn, textsPt } from '../../utils/texts';
import { TextsContext } from '../context/TextsContext';

const baseProps: NavLinksProps = {
    currentRoute: 'services',
    language: 'pt',
    isMobile: false,
    isMenuOpen: false,
};

const renderComponent = (newProps?: Partial<NavLinksProps>): RenderResult => {
    return render(
        <NavLinksContext.Provider
            value={{ isMenuOpen: baseProps.isMenuOpen, toggleMenu: jest.fn() }}
        >
            <NavLinks {...baseProps} {...newProps} />
        </NavLinksContext.Provider>
    );
};

describe('<NavLinks />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const aboutLink = screen.getByText(textsPt.about);
        const projectsLink = screen.getByText(textsPt.projects);
        const servicesLink = screen.getByText(textsPt.services);
        const contactLink = screen.getByText(textsPt.contact);
        const languageButton = screen.getByText('pt');

        expect(aboutLink).toBeInTheDocument();
        expect(projectsLink).toBeInTheDocument();
        expect(servicesLink).toBeInTheDocument();
        expect(contactLink).toBeInTheDocument();
        expect(languageButton).toBeInTheDocument();

        expect(aboutLink).toHaveAttribute('href', `/${baseProps.language}/about`);
        expect(projectsLink).toHaveAttribute('href', `/${baseProps.language}/projects`);
        expect(servicesLink).toHaveAttribute('href', `/${baseProps.language}/services`);
        expect(contactLink).toHaveAttribute('href', `/${baseProps.language}/contact`);

        expect(container).toMatchSnapshot();
    });

    test('renders properly when language is en', () => {
        const { container } = render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <NavLinksContext.Provider
                    value={{ isMenuOpen: baseProps.isMenuOpen, toggleMenu: jest.fn() }}
                >
                    <NavLinks {...baseProps} language="en" />
                </NavLinksContext.Provider>
            </TextsContext.Provider>
        );

        const aboutLink = screen.getByText(textsEn.about);
        const projectsLink = screen.getByText(textsEn.projects);
        const servicesLink = screen.getByText(textsEn.services);
        const contactLink = screen.getByText(textsEn.contact);

        expect(aboutLink).toBeInTheDocument();
        expect(projectsLink).toBeInTheDocument();
        expect(servicesLink).toBeInTheDocument();
        expect(contactLink).toBeInTheDocument();

        expect(aboutLink).toHaveAttribute('href', `/en/about`);
        expect(projectsLink).toHaveAttribute('href', `/en/projects`);
        expect(servicesLink).toHaveAttribute('href', `/en/services`);
        expect(contactLink).toHaveAttribute('href', `/en/contact`);

        expect(container).toMatchSnapshot();
    });

    test('renders properly in mobile', () => {
        const { container } = renderComponent({ isMobile: true });

        expect(container).toMatchSnapshot();
    });

    test('renders properly when the menu is open', () => {
        const { container } = renderComponent({ isMenuOpen: true, isMobile: true });

        expect(container).toMatchSnapshot();
    });
});
