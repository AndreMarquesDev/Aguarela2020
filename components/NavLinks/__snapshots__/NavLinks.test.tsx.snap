// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> renders properly 1`] = `
<div>
  <ul
    class="jsx-1064030420 "
  >
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/about"
      >
        Sobre
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/projects"
      >
        Projetos
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 active"
        href="/pt/services"
      >
        Serviços
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/contact"
      >
        Contacto
      </a>
    </li>
    <li
      class="jsx-1064030420 languageButton"
    >
      <button
        class="jsx-1698987686"
        type="button"
      >
        pt
      </button>
    </li>
  </ul>
</div>
`;

exports[`<NavLinks /> renders properly in mobile 1`] = `
<div>
  <ul
    class="jsx-1064030420 mobileLayout"
  >
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/about"
      >
        Sobre
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/projects"
      >
        Projetos
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 active"
        href="/pt/services"
      >
        Serviços
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/contact"
      >
        Contacto
      </a>
    </li>
    <li
      class="jsx-1064030420 languageButton white"
    >
      <button
        class="jsx-1698987686"
        type="button"
      >
        pt
      </button>
    </li>
  </ul>
</div>
`;

exports[`<NavLinks /> renders properly when language is en 1`] = `
<div>
  <ul
    class="jsx-1064030420 "
  >
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/en/about"
      >
        About
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/en/projects"
      >
        Projects
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 active"
        href="/en/services"
      >
        Services
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/en/contact"
      >
        Contact
      </a>
    </li>
    <li
      class="jsx-1064030420 languageButton"
    >
      <button
        class="jsx-1698987686"
        type="button"
      >
        pt
      </button>
    </li>
  </ul>
</div>
`;

exports[`<NavLinks /> renders properly when the menu is open 1`] = `
<div>
  <ul
    class="jsx-1064030420 mobileLayout menuOpen"
  >
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/about"
      >
        Sobre
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/projects"
      >
        Projetos
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 active"
        href="/pt/services"
      >
        Serviços
      </a>
    </li>
    <li
      class="jsx-1064030420"
    >
      <a
        class="jsx-1064030420 "
        href="/pt/contact"
      >
        Contacto
      </a>
    </li>
    <li
      class="jsx-1064030420 languageButton white"
    >
      <button
        class="jsx-1698987686"
        type="button"
      >
        pt
      </button>
    </li>
  </ul>
</div>
`;
